horovod/common/ops/adasum_mpi_operations.cc

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Microsoft. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =============================================================================
+
+#include "adasum_mpi_operations.h"
+
+namespace horovod {
+namespace common {
+AdasumMPIAllreduceOp::AdasumMPIAllreduceOp(MPIContext* mpi_context, HorovodGlobalState* global_state)
+    : AllreduceOp(global_state), AdasumMPI(mpi_context, global_state) {}
```

Comment:
Compiler warning: change order of parent constructors to `AdasumMPI(mpi_context, global_state), AllreduceOp(global_state)`.